pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ApplicationLifecycle.kt

```diff
@@ -21,30 +21,47 @@ class AppActivityLifecycleCallbacks: Application.ActivityLifecycleCallbacks {
     var startedCount = 0
     var stoppedCount = 0
     fun appInBackground(): Boolean = startedCount <= stoppedCount
+
+    var currentActivity: WeakReference<Activity>? = null
   }
 
-  override fun onActivityPaused(activity: Activity?) {
+  override fun onActivityPaused(activity: Activity) {
+    Companion.stoppedCount +=1
+    currentActivity = null
   }
 
-  override fun onActivityResumed(activity: Activity?) {
+  override fun onActivityResumed(activity: Activity) {
+    currentActivity = WeakReference(activity)
   }
 
-  override fun onActivityStarted(activity: Activity?) {
+  override fun onActivityStarted(activity: Activity) {
     Companion.startedCount += 1
+    currentActivity = WeakReference(activity)
   }
 
-  override fun onActivityDestroyed(activity: Activity?) = Unit
+  override fun onActivityDestroyed(activity: Activity) {
```

Comment:
This method will always be called after `onActivityStopped` so may not be required.

Reply:
That's true. But maybe I'll leave it just in case. It doesn't hurt.